src/Microsoft.Azure.WebJobs.Host.Storage/StorageClientProvider.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using Azure.Core;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Host.Storage
+{
+    /// <summary>
+    /// Abstraction to provide storage clients from the connection names.
+    /// This gets the storage account name via the binding attribute's <see cref="IConnectionProvider.Connection"/>
+    /// property.
+    /// If the connection is not specified on the attribute, it uses a default account.
+    /// </summary>
+    public abstract class StorageClientProvider<TClient, TClientOptions> where TClientOptions : ClientOptions
+    {
+        private readonly AzureComponentFactory _componentFactory;
+        private readonly AzureEventSourceLogForwarder _logForwarder;
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StorageClientProvider{TClient, TClientOptions}"/> class that uses the registered Azure services.
+        /// </summary>
+        /// <param name="configuration">The configuration to use when creating Client-specific objects. <see cref="IConfiguration"/></param>
+        /// <param name="componentFactory">The Azure factory responsible for creating clients. <see cref="AzureComponentFactory"/></param>
+        /// <param name="logForwarder">Log forwarder that forwards events to ILogger. <see cref="AzureEventSourceLogForwarder"/></param>
+        public StorageClientProvider(AzureComponentFactory componentFactory, AzureEventSourceLogForwarder logForwarder, ILogger<StorageClientProvider<TClient, TClientOptions>> logger)
+        {
+            _componentFactory = componentFactory;
+            _logForwarder = logForwarder;
+            _logger = logger;
+
+            _logForwarder.Start();
+        }
+
+        /// <summary>
+        /// Gets the subdomain for the resource (i.e. blob, queue, file, table)
+        /// </summary>
+#pragma warning disable CA1056 // URI-like properties should not be strings
+        protected abstract string ServiceUriSubDomain { get; }
+#pragma warning restore CA1056 // URI-like properties should not be strings
+
+        /// <summary>
+        /// Gets the storage client specified by <paramref name="name"/>
+        /// </summary>
+        /// <param name="name">Name of the connection to use</param>
+        /// <param name="configuration">Configuration to use</param>
+        /// <param name="client">Client that was created if the connection was valid; otherwise, the default value for the type of <paramref name="client"/>.</param>
+        /// <returns>true if the client could be created; false otherwise</returns>
+        public virtual bool TryGet(string name, IConfiguration configuration, out TClient client)
```

Comment:
The "get" naming had me thinking you were doing client caching internally. Since you're always creating a new client, perhaps the naming should be TryCreate/Create?